### PR TITLE
feat(#major);ETH Balance substreams

### PR DESCRIPTION
**Pull request by harsh9200**

### CODE REVIEW

File: eth-balance/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    let mut eth_balance_changes = EthBalanceChanges { items: vec![] };
+
+    for transaction in &block.transaction_traces {
+        for call in &transaction.calls {
+            for balance_change in &call.balance_changes {
+                let new_value = balance_change
+                    .new_value
+                    .as_ref()
+                    .map(|value| {
+                        num_bigint::BigInt::from_bytes_be(num_bigint::Sign::Plus, &value.bytes).into()
```
Comment (this-username-is-taken) [inline eth-balance/src/lib.rs:26]: add a TODO here to replace this with `substreams::scalar::BigInt` when they fix this issue

File: eth-balance/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+use num_bigint;
+use substreams::{store, Hex};
+use substreams::scalar::BigInt;
+use substreams_ethereum::{pb::eth as pbeth};
+use substreams::store::{StoreSetRaw, StoreSet, StoreNew};
+use pb::eth_balance::v1::{EthBalanceChanges, EthBalanceChange};
+
+#[substreams::handlers::map]
+fn map_block_to_balance_changes(
```
Comment (this-username-is-taken) [inline eth-balance/src/lib.rs:14]: Any particular reason of having a mapping module? Why not aggregate directly in the store module?

File: eth-balance/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    block: pbeth::v2::Block
+) ->Result<EthBalanceChanges, substreams::errors::Error> {
+    let mut eth_balance_changes = EthBalanceChanges { items: vec![] };
+
+    for transaction in &block.transaction_traces {
+        for call in &transaction.calls {
+            for balance_change in &call.balance_changes {
```
Comment (this-username-is-taken) [inline eth-balance/src/lib.rs:21]: why not get the balance changes directly from `block.balance_changes`?

Comment (harsh9200) [reply eth-balance/src/lib.rs:21]: `block.balance_changes` only returns the balance change information related to the transaction that mined the block.

File: eth-balance/abi/erc20.json [hunk 4]
```diff
@@ -0,0 +1,222 @@
+[
```
[the remark this answers is not in the document]
Comment (harsh9200) [reply eth-balance/abi/erc20.json:1]: No, I will remove the abi

File: eth-balance/src/lib.rs (hunk 2 above)
Comment (harsh9200) [reply eth-balance/src/lib.rs:14]: No, I just wanted to build a substreams from scratch that will include all the available mapping kinds, I will remove it now and aggregate it directly in the store module

**Review by this-username-is-taken — DISMISSED** [review]
Great work! Let's merge it in

**Pull request merged**